packages/wallet-sdk/src/CoinbaseWalletProvider.ts

```diff
@@ -58,9 +58,13 @@ export class CoinbaseWalletProvider extends ProviderEventEmitter implements Prov
           case 'eth_chainId':
             return hexStringFromNumber(1) as T; // default value
           default: {
-            throw standardErrors.provider.unauthorized(
-              "Must call 'eth_requestAccounts' before other methods"
-            );
+            const signerType = await this.requestSignerSelection(args);
+            const signer = this.initSigner(signerType);
+            await signer.lightHandshake?.(args);
+            this.signer = signer;
+            storeSignerType(signerType);
+
+            return signer.request(args);
```

Comment:
It's strange to default any called RPC into this flow. This assumes intent that all apps are using smart wallet when thats not the case today. For example, calling `eth_accounts` should throw a 4100 err code (not authorized) where this now assumes intent to initialize a smart wallet signer. When walletlink is removed, thats becomes a "safer" assumption but ideally the SDK should remain somewhat agnostic to the connection type ensuring max compatibility in the future.